core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -150,6 +150,10 @@ public UpdateSchema deleteColumn(String name) {
     Preconditions.checkArgument(!updates.containsKey(field.fieldId()),
         "Cannot delete a column that has updates: %s", name);
 
+    if (base != null && base.spec().getFieldsBySourceId(field.fieldId()).size() != 0) {
+      throw new UnsupportedOperationException("Cannot support to drop the partition column: " + name);
```

Comment:
This should be a precondition check like the other validations.

Let's change the error message to `"Cannot drop column used by partition: %s(%s)", partitionField.transform(), name`